subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/daemon/DaemonToolchainIntegrationSpec.groovy

```diff
@@ -0,0 +1,127 @@
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadata
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadataComparator
+import org.gradle.internal.jvm.inspection.JvmVendor
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.gradle.jvm.toolchain.JavaToolchainSpec
+import org.gradle.jvm.toolchain.JvmImplementation
+import org.gradle.jvm.toolchain.JvmVendorSpec
+import org.gradle.jvm.toolchain.internal.JvmInstallationMetadataMatcher
+
+abstract class DaemonToolchainIntegrationSpec extends AbstractIntegrationSpec {
+
+    protected def createDaemonJvmToolchainCriteria(Jvm jvm) {
+        def jvmMetadata = AvailableJavaHomes.getJvmInstallationMetadata(jvm)
+        createDaemonJvmToolchainCriteria(jvmMetadata.languageVersion.majorVersion, jvmMetadata.vendor.knownVendor.name())
+    }
+
+    protected def createDaemonJvmToolchainCriteria(String version = null, String vendor = null, String implementation = null) {
+        def properties = new ArrayList()
+        if (version != null) {
+            properties.add("daemon.jvm.toolchain.version=$version")
+        }
+        if (vendor != null) {
+            properties.add("daemon.jvm.toolchain.vendor=$vendor")
+        }
+        if (implementation != null) {
+            properties.add("daemon.jvm.toolchain.implementation=$implementation")
+        }
+
+        file("gradle/gradle-build.properties") << properties.join(System.getProperty("line.separator"))
+    }
+
+    protected ExecutionResult succeedsSimpleTaskWithDaemonJvm(Jvm expectedDaemonJvm, Boolean daemonToolchainCriteria = true) {
+        return succeedsTaskWithDaemonJvm(expectedDaemonJvm, daemonToolchainCriteria, "help")
+    }
+
+    protected ExecutionResult succeedsTaskWithDaemonJvm(Jvm expectedDaemonJvm, Boolean hasDaemonToolchainCriteria = true, String... tasks) {
+        addDaemonJvmValidation(expectedDaemonJvm, hasDaemonToolchainCriteria)
+        return succeeds(tasks)
+    }
+
+    protected ExecutionResult failsSimpleTaskWithDescription(String expectedExceptionDescription) {
+        def result = fails 'help'
+        failureDescriptionContains(expectedExceptionDescription)
+        return result
+    }
+
+    private def addDaemonJvmValidation(Jvm expectedDaemonJvm, Boolean hasDaemonToolchainCriteria = true) {
+        def expectedJvmMetadata = AvailableJavaHomes.getJvmInstallationMetadata(expectedDaemonJvm)
+        def expectedVendor = expectedJvmMetadata.vendor
+        def expectedJavaHome = expectedJvmMetadata.javaHome
+        def expectedVersion = expectedJvmMetadata.javaVersion
+        if (hasDaemonToolchainCriteria) {
+            // When Daemon toolchain criteria is defined the detection mechanism should be taken into consideration
+            // for the resolution of the toolchain that may be different depending on the local installed ones
+            def detectedMatchingToolchain = findMatchingToolchain(expectedDaemonJvm.javaVersion, expectedVendor)
```

Comment:
The main point here is the fact that those tests are validating the E2E flow, validating the resolved Daemon toolchain based on the defined criteria. Meaning that in case there are multiple toolchain 17 installed locally we need to validate against the one we expected to be resolved based on [precedence rule](https://docs.gradle.org/current/userguide/toolchains.html#sec:precedence). For this reason is required if the criteria is defined to use the `JvmInstallationMetadataComparator` which is already tested separately to guarantee the expected result.

As pointed out, the `org.gradle.java.installations.paths` option doesn't apply for Daemon toolchain, for this reason we cannot disable the `auto-detection` mechanism and relay that this will be resolved always with the specified one.